app/cluster-api-provider-sidero/controllers/metalmachine_controller.go

```diff
@@ -214,7 +214,7 @@ func (r *MetalMachineReconciler) reconcileDelete(ctx context.Context, metalMachi
 			return ctrl.Result{}, err
 		}
 
-		serverResource.Status.InUse = false
+		serverResource.Status.NeedsWipe = true
```

Comment:
We keep the server marked as in use as a way to prevent it from being mistakenly allocated again. We should only mark it as not in use once the reset process is done.